### PR TITLE
fix: Make `cargo install --git` work with the repo

### PR DESCRIPTION
**Pull request by shamilsan**
It is unable to run `cargo install --git https://github.com/gear-tech/gear.git <some-gear-crate>`.

Before: 
```
❯ cargo install --locked --git https://github.com/gear-tech/gear.git wasm-proc
    Updating git repository `https://github.com/gear-tech/gear.git`
error: package `/Users/shamil/.cargo/git/checkouts/gear-bdc2b7d48ef3acbd/32c1b91/examples/binaries/new-meta/io/Cargo.toml` is a member of the wrong workspace
expected: /Users/shamil/.cargo/git/checkouts/gear-bdc2b7d48ef3acbd/32c1b91/Cargo.toml
actual:   /Users/shamil/.cargo/git/checkouts/gear-bdc2b7d48ef3acbd/32c1b91/examples/Cargo.toml
```

After:
```
❯ cargo install --locked --git https://github.com/gear-tech/gear.git --branch sy-fix-cargo-install wasm-proc
    Updating git repository `https://github.com/gear-tech/gear.git`
  Installing wasm-proc v0.1.0 (https://github.com/gear-tech/gear.git?branch=sy-fix-cargo-install#2925afcf)
         ... ...
    Finished release [optimized] target(s) in 1m 21s
  Installing /Users/shamil/.cargo/bin/wasm-proc
   Installed package `wasm-proc v0.1.0 (https://github.com/gear-tech/gear.git?branch=sy-fix-cargo-install#2925afcf)` (executable `wasm-proc`)
```

@gear-tech/dev

### CODE REVIEW

**Review by DennisInSky — APPROVED** [review]

File: examples/Cargo.toml [hunk 1]
```diff
@@ -46,7 +46,11 @@ members = [
 ]
 
 exclude = [
-    "binaries/*"
+    "binaries/*",
+    "binaries/new-meta/io",
+    "binaries/new-meta/state-v1",
+    "binaries/new-meta/state-v2",
+    "binaries/new-meta/state-v3",
```
Comment (breathx) [inline examples/Cargo.toml:53]:
```suggestion
    "binaries/**",
```
[ref](https://docs.rs/glob/0.3.0/glob/struct.Pattern.html)

**Review by breathx — CHANGES_REQUESTED** [review]
Why did locks update?

File: examples/Cargo.toml (hunk 1 above)
Comment (shamilsan) [reply examples/Cargo.toml:53]: I've tried `binaries/**`, `binaries/**/*`, `binaries/new-meta/*`, `binaries/new-meta/**` etc. but nothing worked to me.

**Comment by shamilsan** [comment]
> Why did locks update?

Just in case 😄

**Comment by shamilsan** [comment]
@breathx Kindly reminder to re-review. Globs don't work for me, so I've excluded several crates from the workspace explicitly.

**Comment by shamilsan** [comment]
This will be fixed after #2602. So I am closing it at the moment.

**Comment by shamilsan** [comment]
Better alternative: #2885

**Pull request closed without merging**